google/cloud/pubsub/admin/subscription_admin_client.h

```diff
@@ -0,0 +1,925 @@
+  ///
+  /// [Protobuf mapping rules]: https://protobuf.dev/reference/cpp/cpp-generated/
+  /// [input iterator requirements]: https://en.cppreference.com/w/cpp/named_req/InputIterator
+  /// [`std::string`]: https://en.cppreference.com/w/cpp/string/basic_string
+  /// [`future`]: @ref google::cloud::future
+  /// [`StatusOr`]: @ref google::cloud::StatusOr
+  /// [`Status`]: @ref google::cloud::Status
+  /// [google.pubsub.v1.ListSnapshotsRequest]: @googleapis_reference_link{google/pubsub/v1/pubsub.proto#L1518}
+  /// [google.pubsub.v1.Snapshot]: @googleapis_reference_link{google/pubsub/v1/pubsub.proto#L1476}
+  ///
+  // clang-format on
+  StreamRange<google::pubsub::v1::Snapshot>
+  ListSnapshots(google::pubsub::v1::ListSnapshotsRequest request, Options opts = {});
+
+  // clang-format off
+  ///
+  /// Creates a snapshot from the requested subscription. Snapshots are used in
+  /// [Seek](https://cloud.google.com/pubsub/docs/replay-overview) operations,
+  /// which allow you to manage message acknowledgments in bulk. That is, you can
+  /// set the acknowledgment state of messages in an existing subscription to the
+  /// state captured by a snapshot.
+  /// If the snapshot already exists, returns `ALREADY_EXISTS`.
+  /// If the requested subscription doesn't exist, returns `NOT_FOUND`.
+  /// If the backlog in the subscription is too old -- and the resulting snapshot
+  /// would expire in less than 1 hour -- then `FAILED_PRECONDITION` is returned.
+  /// See also the `Snapshot.expire_time` field. If the name is not provided in
+  /// the request, the server will assign a random
+  /// name for this snapshot on the same project as the subscription, conforming
+  /// to the [resource name format]
+  /// (https://cloud.google.com/pubsub/docs/admin#resource_names). The
+  /// generated name is populated in the returned Snapshot object. Note that for
+  /// REST API requests, you must specify a name in the request.
+  ///
+  /// @param name  Required. User-provided name for this snapshot. If the name is not provided
+  ///  in the request, the server will assign a random name for this snapshot on
+  ///  the same project as the subscription. Note that for REST API requests, you
+  ///  must specify a name.  See the [resource name
+  ///  rules](https://cloud.google.com/pubsub/docs/admin#resource_names). Format
+  ///  is `projects/{project}/snapshots/{snap}`.
+  /// @param subscription  Required. The subscription whose backlog the snapshot retains.
+  ///  Specifically, the created snapshot is guaranteed to retain:
+  ///   (a) The existing backlog on the subscription. More precisely, this is
+  ///       defined as the messages in the subscription's backlog that are
+  ///       unacknowledged upon the successful completion of the
+  ///       `CreateSnapshot` request; as well as:
+  ///   (b) Any messages published to the subscription's topic following the
+  ///       successful completion of the CreateSnapshot request.
```

Comment:
FYI, some of this documentation may not render as well as we would like.  Nothing we can do, maybe just look at it and then ask or send fixes upstream.

Reply:
I would like it to be rendered as the service team intended.  I this case I guess they wanted an enumerated list, but I am not sure markdown will represent it as such.  I think most markdown engines will treat that text as a single sentence.

Proto comments are supposed to be CommonMark:

https://google.aip.dev/192#formatting

But not all service teams follow that guidance, and the results are hilarious sometimes.  This one in particular does not look like CommonMark, but I might be wrong.

PS: even if it is or becomes formatted in CommonMark "standard": the C++ tooling and/or the tooling for other languages may mess things up as it goes through 1 or 2 markdown engines that may not perfectly follow the  CommonMark guidelines in all cases.

Reply:
Hm.. my interpretation was this wasn't markdown at all. But they are using [backticks](https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto#L1440-L1448)...

We can see how it looks and try and fix later maybe, but sounds like change is not blocking. SGTM!